fix license file to conform to MIT rules, add License section to README (#645)
stac-utils/stac-server: README.md

## About

[stac-server](https://github.com/stac-utils/stac-server) was forked from [sat-api](https://github.com/sat-utils/sat-api). Stac-server is for STAC versions 0.9.0+, while sat-api exists for versions of STAC prior to 0.9.0.

## License

> stac-server is licensed under [The MIT License](https://opensource.org/license/mit/).
> Copyright for portions of stac-server are held by Development Seed (2016) as
> part of project [sat-api](https://github.com/sat-utils/sat-api)
> [original license](https://github.com/sat-utils/sat-api/blob/master/LICENSE). All other
> copyright for stac-server is held by Element 84, Inc (2020).

Reviewer: ```suggestion
stac-server is licensed under [The MIT License](https://opensource.org/license/mit/).
Copyright for portions of stac-server is held by Development Seed (2016) as
part of project [sat-api](https://github.com/sat-utils/sat-api)
[original license](https://github.com/sat-utils/sat-api/blob/master/LICENSE). Copyright for all changes to stac-server since the fork date is held by Element 84, Inc (2020).
```
